src/System.Private.ServiceModel/src/System/ServiceModel/Channels/BufferedWriteStream.cs

```diff
@@ -57,12 +60,12 @@ public BufferedWriteStream(Stream stream, int bufferSize)
         private void EnsureNotClosed()
         {
             if (_stream == null)
-                throw new ObjectDisposedException("BufferedWriteStream");
+                throw new ObjectDisposedException(nameof(BufferedWriteStream));
         }
 
         private void EnsureCanWrite()
         {
-            Contract.Requires(_stream != null);
+            Contract.Assert(_stream != null);
             if (!_stream.CanWrite)
                 throw new NotSupportedException("write");
```

Comment:
I know this message shouldn't happen... that said, should we fix the exception message here?